ouroboros-consensus/src/Ouroboros/Consensus/Mempool/API.hs

```diff
@@ -234,63 +219,43 @@ isMempoolTxRejected :: MempoolAddTxResult blk -> Bool
 isMempoolTxRejected MempoolTxRejected{} = True
 isMempoolTxRejected _                   = False
 
--- | Wrapper around 'implTryAddTxs' that blocks until all transaction have
--- either been added to the Mempool or rejected.
+-- | A wrapper around 'addTx' that adds a sequence of transactions on behalf of
+-- a remote peer.
 --
--- This function does not sync the Mempool contents with the ledger state in
--- case the latter changes, it relies on the background thread to do that.
+-- Note that transactions are added one by one, and can interleave with other
+-- concurrent threads using 'addTx'.
 --
--- See the necessary invariants on the Haddock for 'tryAddTxs'.
+-- See 'addTx' for further details.
 addTxs
   :: forall m blk. MonadSTM m
   => Mempool m blk
   -> [GenTx blk]
   -> m [MempoolAddTxResult blk]
-addTxs mempool = addTxsHelper mempool DoNotIntervene
+addTxs mempool = mapM (addTx mempool AddTxForRemotePeer)
 
--- | Variation on 'addTxs' that is more forgiving when possible
+-- | A wrapper around 'addTx' that adds a sequence of transactions on behalf of
+-- a local client. This reports more errors for local clients, see 'Intervene'.
+--
+-- Note that transactions are added one by one, and can interleave with other
+-- concurrent threads using 'addTx'.
 --
--- See 'Intervene'.
+-- See 'addTx' for further details.
 addLocalTxs
   :: forall m blk. MonadSTM m
   => Mempool m blk
   -> [GenTx blk]
   -> m [MempoolAddTxResult blk]
-addLocalTxs mempool = addTxsHelper mempool Intervene
+addLocalTxs mempool = mapM (addTx mempool AddTxForLocalClient)
+
+-- | Who are we adding a tx on behalf of, a remote peer or a local client?
+--
+-- This affects two things:
+--
+-- 1. how certain errors are treated: we want to be helpful to local clients.
+-- 2. priority of service: local clients are prioritised over remote peers.
```

Comment:
I wonder if we should explain this a bit further. In particular, could the reader misunderstand this and think that local clients can starve remote peers (which is not the case as far as I understand). 

I have trouble defining this kind of priority. Could we say something like: given there is at least one remote peer and one local client waiting to add a transaction if a remote peer and a local client try to add additional transactions, then the one from the local client will be added first?

Reply:
I've added a patch with some more detailed documentation of the fairness property: both what the mempool API should provide, and the specific strategy that the current implementation uses.

I've defined it in terms of share of the resource that the threads end up getting.